apps/full-site-editing/full-site-editing-plugin/premium-content/blocks/logged-out-view/edit.js

```diff
@@ -23,13 +23,17 @@ import Context from '../container/context';
  * @property { string } containerClientId
  * @property { Attributes } attributes
  * @property { Function } setAttributes
- * @property { Function } selectBlock
+ * @property { Function } selectContainerBlock
  *
  * @param { Props } props Properties
  */
-function Edit( { selectBlock } ) {
+function Edit( { selectContainerBlock } ) {
 	useEffect( () => {
-		selectBlock();
+		// Selects the container block on mount.
+		//
+		// Execution delayed with setTimeout to ensure it runs after any block auto-selection performed by inner blocks
+		// (such as the Recurring Payments block). @see https://github.com/Automattic/wp-calypso/issues/43450
+		setTimeout( selectContainerBlock, 0 );
```

Comment:
I think this can cause a flash of the block title in the sidebar from "Recurring Payments" to "Premium Content", but it doesn't sound like anything to be dealt with better here 👍

Reply:
I think this is fine for now. As a follow up we could have the child not dispatch the selection event when we detect we're nested within particular containers. 

If we see tricky behavior like this in other spots, it's also possible to create a custom store to better manage shared state between payment blocks.

Reply:
What's interesting with the serialized blocks here is that it's possible for a mismatch to occur between the premium content block and subscription button (say we edit this manually, or a save is botched for some reason).

Just thinking out loud, it'd eventually be more ideal to not be possible to save this sort of nonsense state, though offhand I'm not sure what the best serialized shape for that would be. Maybe compose a block that isn't rendered that represents a plan? Allow planId of inherit? No need to follow up on this, but wanted to leave a note in case we run into similar situations elsewhere.

Notice the two instance of planId "164".
```
<!-- wp:premium-content/container {"selectedPlanId":164} -->
<div class="wp-block-premium-content-container">
	<!-- wp:premium-content/subscriber-view -->
	<div class="wp-block-premium-content-subscriber-view">
		<!-- wp:paragraph {"placeholder":"Insert the piece of content you want your visitors to see after they subscribe."} -->
			<p></p>
		<!-- /wp:paragraph --></div>
	<!-- /wp:premium-content/subscriber-view -->

	<!-- wp:premium-content/logged-out-view -->
	<div class="wp-block-premium-content-logged-out-view">
		<!-- wp:heading {"level":3} -->
		<h3>Subscribe to get access</h3>
		<!-- /wp:heading -->

		<!-- wp:paragraph -->
		<p>Read more of this content when you subscribe today.</p>
		<!-- /wp:paragraph -->

		<!-- wp:premium-content/buttons -->
		<div class="wp-block-premium-content-buttons wp-block-buttons">
			<!-- wp:jetpack/recurring-payments {"planId":164,"submitButtonText":"Subscribe"} /-->
		
			<!-- wp:premium-content/login-button -->
			<div class="wp-block-premium-content-login-button wp-block-button"><a class="wp-block-button__link">Log in</a></div>
			<!-- /wp:premium-content/login-button --></div>
		<!-- /wp:premium-content/buttons -->

	<!-- /wp:premium-content/logged-out-view --></div>
<!-- /wp:premium-content/container -->
```